modules/containers/include/hephaestus/containers/blocking_queue.h

```diff
@@ -25,12 +25,19 @@ concept SimilarTo = std::same_as<std::remove_cvref_t<T>, std::remove_cvref_t<U>>
 template <class T>
 class BlockingQueue {
 public:
-  /// Create a queue
+  /// Create a queue, leave constructor public to go around throwing input parameter checks.
```

Comment:
mmm I think we should remove the constructor, it is a weird pattern like this, nobody would use `create`